### PR TITLE
Allow skydns itself to use env variables

### PR DESCRIPTION
**Pull request by miekg**
Reuse the variables defined in skydnsctl for skydns itself. Also rename
them slightly to keep the two much more in sync.

SKYDNS_DNSPORT -> SKYDNS_DNS (something like 127.0.0.1:1053) in stead of a number
SKYDNS_DNSDOMAIN -> SKYDNS_DOMAIN (a string like skydns.local)

The environment variable SKYDNS is used in skydns, the prefix http:// is
removed before use. This might confuse people.

Sadly this is a minor backward incompatible change, because
skydns.NewClient needed to be changed

### CODE REVIEW

File: client/client.go [hunk 1]
```diff
@@ -184,10 +176,10 @@ func (c *Client) GetAllServicesDNS() ([]*msg.Service, error) {
 		if v, ok := r.(*dns.SRV); ok {
 			s[i] = &msg.Service{
 				// TODO(miek): uehh, stuff it in Name?
-				Name: v.Header().Name + " (Priority: " + strconv.Itoa(int(v.Priority)) + ", " + "Weight: " + strconv.Itoa(int(v.Weight)) +")",
+				Name: v.Header().Name + " (Priority: " + strconv.Itoa(int(v.Priority)) + ", " + "Weight: " + strconv.Itoa(int(v.Weight)) + ")",
```
Comment (crosbymichael) [inline client/client.go:179]: While you are doing some refactoring do you think that `fmt.Sprintf` should be more readable for things like this?

Comment (miekg) [inline client/client.go:179]: [ Quoting notifications@github.com in "Re: [skydns] Allow skydns itself to..." ]

> > @@ -184,10 +176,10 @@ func (c _Client) GetAllServicesDNS() ([]_msg.Service, error) {
> >         if v, ok := r.(*dns.SRV); ok {
> >             s[i] = &msg.Service{
> >                 // TODO(miek): uehh, stuff it in Name?
> > -               Name: v.Header().Name + " (Priority: " + strconv.Itoa(int(v.Priority)) + ", " + "Weight: " + strconv.Itoa(int(v.Weight)) +")",
> > -               Name: v.Header().Name + " (Priority: " + strconv.Itoa(int(v.Priority)) + ", " + "Weight: " + strconv.Itoa(int(v.Weight)) + ")",
> 
> While you are doing some refactoring do you think that `fmt.Sprintf` should be more readable for things like this? 

hadn't thought about using Sprintf here, but yes, your completely right.

But this should be done in another PR and we should talk about the output,
because right now it is put in the Name which is kinda of ugly.

## 

   Miek Gieben

Comment (crosbymichael) [inline client/client.go:179]: Sounds good to me

**Comment by bketelsen** [comment]
lgtm  @crosbymichael Ok by you?

**Comment by crosbymichael** [comment]
LGTM
